### PR TITLE
Tags

### PR DESCRIPTION
**Pull request by IvarStefansson**
Main change: 
Translate face-tags into less elegant but more general form. Now geometry tags are found as e.g.
g.tags['boundary_faces']
and other information should be stored in the data corresponding to g (g.node_props...). Other tags, e.g. node tags for the geometry should be in the same format (g.tags['domain_boundary_nodes'])!

Also, I updated the BCs slightly. The discretizations are now allowed to override the type on internal boundaries.

### CODE REVIEW

File: src/porepy/grids/grid.py [hunk 1]
```diff
@@ -572,7 +553,7 @@ def get_internal_faces(self):
             np.ndarray (1d), index of internal faces.
 
         """
-        return self.__indices(self.has_not_face_tag(FaceTag.BOUNDARY))
+        return self.__indices(np.logical_not(self.tags['bound_faces']))
```
Comment (rbe051) [inline src/porepy/grids/grid.py]: The tag should be 'boundary_faces' ?

File: src/porepy/utils/tags.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+"""
+Methods for tag handling. The following primary applications are intended:
+    --Grid tags, stored in the grids and data fields of the grid bucket.
+    Geometry tags are stored in the grids, typical fields
+    are cell, face or node tags, lists of length g.num_cell etc. Other
+    information is storad in the data fields. The entries
+    of the lists should be boolean or integers. Examples:
+        g.tags['fracture_faces'] = [0, 1, 1, 0, 1, 1]
+        g.tags['fracture_face_ids'] = [0, 1, 2, 0, 1, 2]
```
Comment (rbe051) [inline src/porepy/utils/tags.py]: the fracture_face_ids should be: [1, 2, 4, 5]

File: src/porepy/utils/tags.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    for a grid with two immersed fractures (neighbour to faces (1 and 4) and
+    (2 and 5), respectively). If the wells are located in cells 1 and 3, this
+    may be tagged in the data as e.g.
+        data['well_cells'] = [0 1 0 -1]
```
Comment (rbe051) [inline src/porepy/utils/tags.py]: should be [0, 1, 0, **1**] ?

File: src/porepy/utils/tags.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    --Fracture network tags, stored in the fracture network field .tags. One
+    list entry for each fracture:
+        network.tags['fracture_id'] = [1,2]
+    --
+"""
+import numpy as np
+
+
+def append_tags(tags, kws, appendices):
+    for i, kw in enumerate(kws):
```
Comment (rbe051) [inline src/porepy/utils/tags.py]: missing docstring

File: src/porepy/utils/tags.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        tags[kw] = np.append(tags[kw], appendices[i])
+
+
+def standard_face_tags():
+    keys = ['fracture_faces', 'boundary_faces',
```
Comment (rbe051) [inline src/porepy/utils/tags.py]: also here

File: test/integration/test_advective_diffusive.py [hunk 6]
```diff
@@ -142,15 +141,13 @@ def __init__(self, g, physics='transport'):
         SourceAdvectiveDiffusiveProblem.__init__(self, g, physics=physics)
 
     def bc(self):
-        dir_faces = self.grid().has_face_tag(FaceTag.DOMAIN_BOUNDARY)
-        dir_faces = np.ravel(np.argwhere(dir_faces))
+        dir_faces = self.grid().get_domain_boundary_faces()
```
Comment (rbe051) [inline test/integration/test_advective_diffusive.py]: should this function be removed and replaced by using g.tags['domain_boundary_faces') ??

File: examples/example1/benchmark1/test_fv.py [hunk 7]
```diff
@@ -44,33 +44,34 @@ def add_data(gb, domain, kf, mesh_value):
         param.set_source('flow', np.zeros(g.num_cells))
 
         # Boundaries
-        bound_faces = g.get_boundary_faces()
-        if bound_faces.size == 0:
-            continue
+        bound_faces = g.get_domain_boundary_faces()
```
[the remark this answers is not in the document]
Comment (keileg) [reply examples/example1/benchmark1/test_fv.py]: Yes!

File: src/porepy/grids/grid.py (hunk 1 above)
Comment (keileg) [reply src/porepy/grids/grid.py]: I agree

File: examples/example1/benchmark1/test_fv.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (keileg) [reply examples/example1/benchmark1/test_fv.py]: I agree

File: src/porepy/fracs/importer.py [hunk 8]
```diff
@@ -392,51 +400,53 @@ def read_dfn_grid(folder, num_fractures, case_id, **kwargs):
...
-        nodes_2d, face_nodes_2d, cell_faces_2d = _dfn_grid_2d(folder, post, **kwargs)
+        post = "_F" + str(f_id + offset_name) + "_" + str(case_id) + ".txt"
+        nodes_2d, face_nodes_2d, cell_faces_2d = _dfn_grid_2d(
+            folder, post, **kwargs)
         g_2d[f_id] = grid.Grid(2, nodes_2d, face_nodes_2d, cell_faces_2d,
                                "fracture_" + str(f_id) + "_" + str(case_id))
 
         bnd_faces = g_2d[f_id].get_boundary_faces()
-        g_2d[f_id].add_face_tag(bnd_faces, grid.FaceTag.DOMAIN_BOUNDARY)
+        g_2d[f_id].tags['domain_boundary_faces'][bnd_faces] = True
```
[the remark this answers is not in the document]
Comment (keileg) [reply src/porepy/fracs/importer.py]: I agree with all_boundary to get everything, it forces me to think twice about what boundary am I looking for (just the word boundary soon leads you to take a wrong shortcut).
In that spirit, I think I prefer keeping domain_boundary, but I'm open to other options.

File: src/porepy/grids/grid.py [hunk 9]
```diff
@@ -726,31 +708,11 @@ def min_dist(pts):
             ci[i] = min_dist(p[:, i].reshape((3, -1)))
         return ci
 
-    def add_face_tag(self, f, tag):
```
Comment (keileg) [inline src/porepy/grids/grid.py]: I will not miss these!

**Pull request merged**